recursion_problem.rb

```diff
@@ -0,0 +1,118 @@
+# #Recursion Program On:
+
+#reverse a string
+
+def reverse_string(str)
+  a = str.size < 2 ? str : str[-1] + reverse_string(str[1..-2]) + str[0]
+  a = a.split("")
+  a.delete_if { |element| "aeiou".include?(element) }
+  string = a.join("")
+end
+
+p reverse_string("abcdefghi")
+
+#sum a string
+
+def sum_num(arr)
```

Comment:
Getting `TypeError (String can't be coerced into Integer)` on running this code!

Reply:
Sorted